### PR TITLE
add Cython version eval_market1501_wrap

### PR DESCRIPTION
**Pull request by luzai**
Thanks a lot for your repo!

Ref to issue #15. I make some modification:
- Fix some bugs: 
``` python
# When any gallery image match the query identity, mark orig_cmc_flag as 1. 
if matches[q_idx][idx]>1e-31:
    orig_cmc_flag=1
# When query identity does not appear in gallery, mark corresponding AP as -1. Do not count into mAP.
if not orig_cmc_flag:
            all_AP[q_idx]=-1
            continue
```
- Add `eval_lib/test.py` to test whether two version eval_market1501 equivalent.  

I am using python3. To run `python test.py` from terminal, we may need add the following code to the beginning of `test.py`.

``` python 
from __future__ import absolute_import
import sys, os

sys.path.insert(
    0, os.path.dirname(os.path.abspath(__file__)) + '/..'
)
```

And my outputs is 
image

There are still slight difference between two version, which can be ignore under some tolerance, but I still wonder where the difference comes from.

### CODE REVIEW

File: eval_metrics.py [hunk 1]
```diff
@@ -115,7 +116,10 @@ def eval_market1501(distmat, q_pids, g_pids, q_camids, g_camids, max_rank):
         tmp_cmc = orig_cmc.cumsum()
         tmp_cmc = [x / (i+1.) for i, x in enumerate(tmp_cmc)]
         tmp_cmc = np.asarray(tmp_cmc) * orig_cmc
-        AP = tmp_cmc.sum() / num_rel
+        if num_rel !=0:
```
Comment (KaiyangZhou) [inline eval_metrics.py]: Line103 already checks `orig_cmc `.

**Comment by luzai** [comment]
For run `test.py` but got error ImportError: No module named eval_lib.cython_eval: 
we need add a `__init__.py` to `eval_lib/` for python2, so that eval_lib is a package. 

The modification in new commit include: 
- remove duplicate `if num_rel !=0`
- `train_img_model_xent_dev.py` to compare two version evaluation code
- Others is mainly for compatibility between python3 and python2

run `python setup.py -i` under `eval_lib/`, and run `python train_img_model_xent_dev.py` with correct `--resume` and `--dataset` option, we may get something like:
image

**Comment by KaiyangZhou** [comment]
Thank you so much. I merged to the dev branch. Will update to master soon.

**Comment by KaiyangZhou** [comment]
now all set

**Pull request merged**